src/app/modules/submission/email-submission/email-submission.util.ts

```diff
@@ -484,20 +484,26 @@ const createFormattedDataForOneField = <T extends EmailDataFields | undefined>(
     response: ResponseFormattedForEmail,
     hashedFields: Set<MyInfoKey>,
   ) => T,
-): T[] => {
+  childIdx: number,
```

Comment:
createFormattedDataForOneField

Reply:
went with the 2nd approach. thanks for the suggestion, it's a lot cleaner now! [c4cb6a4](https://github.com/opengovsg/FormSG/pull/6745/commits/c4cb6a4b2eaa79b8ac649a110de7abb9c4d779c0)